### PR TITLE
fix sprite frame cache, and remove unnecessary headers dependencies.

### PR DESCRIPTION
**Pull request by joewan**
we should add private headers for ccs, do not expose privae headers to users.

### CODE REVIEW

**Comment by joewan** [comment]
at present, it will cause memory leak. I am working for it, pls do not merge.

**Comment by joewan** [comment]
11.07
memory leak:If I create two armatures, and then delete them, unload the resources, we can't release the texture any more.

11.18
the message of 11.07 is wrong, now there is no above problems.

**Comment by joewan** [comment]
@minggo @chengstory please help review this pr.
1.when we load and create armature, both keep the reference to the used texture and sprite frames.

step1:load and create an armature;
step2:director purge cache data
step3:create an armature again, we can not show this armature because the sprite frame has been purged.

2.fix memory leak
3.remove unnecessary headers dependencies.

**Comment by minggo** [comment]
@chengstory 
Please take a look.

**Comment by chengstory** [comment]
@2youyouo2 Please take a look

File: cocos/editor-support/cocostudio/CCSpriteFrameCacheHelper.h [hunk 1]
```diff
@@ -50,11 +56,16 @@ class CC_STUDIO_DLL SpriteFrameCacheHelper
      *	@brief	Add sprite frame to CCSpriteFrameCache, it will save display name and it's relative image name
      */
     void addSpriteFrameFromFile(const std::string& plistPath, const std::string& imagePath);
+    void removeSpriteFrameFromFile(const std::string& plistPath);
 
 private:
+    void _retainSpriteFrames(const std::string& plistPath);
```
Comment (2youyou2) [inline cocos/editor-support/cocostudio/CCSpriteFrameCacheHelper.h]: Thanks for your work!
But function should not be named with "_" even it's a private function.
You can find the cocos coding style here: 
https://github.com/cocos2d/cocos2d-x/blob/v3/docs/CODING_STYLE.md#function-parameter-ordering

**Comment by joewan** [comment]
@2youyouo2 pls review again, thanks.

**Comment by 2youyou2** [comment]
I think it's ok, thanks.

**Pull request merged**